tools/src/main/scala/scala/scalanative/tools/Config.scala

```diff
@@ -1,80 +1,197 @@
 package scala.scalanative
 package tools
 
-import java.io.File
+import java.nio.file.{Path, Paths}
+
 import nir.Global
 
 sealed trait Config {
 
+  /** Path to the nativelib jar. */
+  def nativeLib: Path
+
+  /** The driver to use for the optimizer. */
+  def driver: OptimizerDriver
+
+  /** The reporter that receives messages from the linker. */
+  def linkerReporter: LinkerReporter
+
+  /** The reporter that receives messages from the optimizer. */
+  def optimizerReporter: OptimizerReporter
```

Comment:
Having pluggable `OptimizerReporter` and `LinkerReporter` leaks NIR into the build API. We need to replace with something less powerful. For example: we can add a boolean option that turns on an off current to-file reporters.

Reply:
The `Reporter`s no longer leak NIR.